### ISSUE TITLE
Shadows behave weirdly during animations

### ISSUE THREAD

**Issue by dccsillag** [dccsillag/picom#2]
During animations, the window shadows will generally follow the size animation precisely. But it appears that this may not always be the case, and even when it is, it may not look very good (because size transitions don't interpolate the window contents).

A possible workaround (credits to @podiki, https://github.com/dccsillag/picom/issues/1#issuecomment-877735114) is to simply disable shadows during animations. But this may look weird or not polished.

It's probably a good idea to come back to this only after we have window content interpolation (https://github.com/dccsillag/picom/issues/3).

**Comment by dccsillag** [comment]
Now that #3 is somewhat done, it is clear that shadows behave quite normally. It was just that there was empty space above the shadows, without interpolating content.

Closing this issue.

**Comment by podiki** [comment]
Just a note regarding shadows, when enabling still get unwanted behavior. Not sure if you want a new issue or in one of the others.

Thankfully it is not windows flying around and unusable animations, but instead the transitions look nearly instantaneous without the actual animation (occasionally you get part of an animation). There is a slight artifact on/shudder/something on some of the shadows when there is a transition. Bit hard to explain, but normal behavior when shadows are disabled.

**Comment by dccsillag** [comment]
I've just played around with this a bit, and I think I see it (though I don't think it's quite as severe here :/).

I think this is a performance problem -- that is, with shadows, the redraw takes so much longer that we pretty much "skip" the whole animation. In fact, if you lower the animation stiffness, you should be able to see more of the animation.

It's quite curious that simply enabling shadows causes this much of a performance hit.

Could you take a look at whether this also happens before the commit adding window content animation? (Also, what backend are you using? I'm guessing `--experimental-backends --backend glx`.)

If you can confirm that on your end this also seems like a performance problem, could you open a separate issue explicitely mentioning that we are dealing with something that's performance-related? Thanks.

**Comment by podiki** [comment]
I've just tried this on my desktop (Nvidia graphics, previously was on a laptop using the Intel integrated graphics) and it works fine even with shadows, same config file. Both are run with `--experimental-backends` and have `backend` set to `glx`. So then likely performance issue, or something with the Intel driver (I've had some issues with it in the past too). I didn't try with previous commits, and I think I changed the animation stiffness with no effect though.

I will open a new issue, what info would be useful besides what I've said so far?

**Comment by dccsillag** [comment]
I can't think of any more useful info, but maybe something will pop up after proper "debugging" has begun.

I'm not sure it's necessarily related to the Intel driver; although I use the glx backend here, my graphics card is in modesetting mode, and picom ends up just using the intel integrated graphics instead -- and I don't have much of this problem. But it's something to keep in mind.

To be more precise, I've been able to reproduce this only by having both my built-in FHD laptop screen and 4k external monitor connected. So I think it might just be some performance difference between our integrated graphics.
